igcollect/sfp_dom.py

```diff
@@ -251,26 +257,35 @@ def ethtool_get_module_data(interface):
     try:
         fcntl.ioctl(sock, SIOCETHTOOL, ifr)
     except OSError as exc:
-        if exc.errno == 95:
-            raise NoSFPException(f'Interface {interface} has no SFP module') \
+        # Cumulus will return OSError 5 when you query for eth0, eth1
+        # Debian will return OSError 95 for interfaces without SFP
+        if exc.errno == 5 or exc.errno == 95:
+            raise NoSFPException(f'{interface}: SFP not present') \
                 from exc
         else:
             raise
 
     # Build the struct with the ETHTOOL_GMODULEEEPROM command and request.
     # Objects below are ctypes structs that will be later used with the
     # ioctl calls.
-    ee = EthtoolEEPROM(cmd=ETHTOOL_GMODULEEEPROM, len=em.eeprom_len, offset=0)
+    # We limit the length to 604, as that is the biggest size of an EEPROM
+    # according to the standard and we don't want buffer overruns.
+    length = 604 if em.eeprom_len > 604 else em.eeprom_len
```

Comment:
`min(604, em.eeprom_len)` would be a bit more elegant, and while at it make `604` a constant maybe

Reply:
Both done :)